### PR TITLE
Search users by tags

### LINKED ISSUE #2323: [FEATURE] search by clicking on user's tags
### Description

Currently we can add tags to our profiles, but it would be good to be able to search for these and also by clicking them it goes to the search page with it pre-filled 

### Screenshots

_No response_

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by joeshajan**
## Fixes Issue

Closes #2323 

## Changes proposed

- [x] searching users by tags when the tag name is entered in the search input.
- [x] redirecting to the search page and searching for users by tag when clicking on any tag in any of the user's profiles.

## Check List (Check all the applicable boxes) 

- [x] My code follows the code style of this project.
- [ ] My change requires changes to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] All new and existing tests passed.
- [x] This PR does not contain plagiarized content.
- [x] The title of my pull request is a short description of the requested changes.

## Screenshots

**Results displayed after typing tag in input**

image

**Redirect to the search page when clicking on any tag**

image

**Tag name is passed as query params when redirecting to the search page and tag name is auto-filled in the input field**

image

<a href="https://gitpod.io/#https://github.com/EddieHubCommunity/LinkFree/pull/2341"></a>

### CODE REVIEW

File: pages/search.js [hunk 1]
```diff
@@ -44,9 +45,14 @@ export default function Search({ users }) {
...
-      results = users.filter((user) =>
-        user.name.toLowerCase().includes(value.toLowerCase())
-      );
+      results = users.filter((user) => {
+        if (user.name.toLowerCase().includes(value.toLowerCase())) return true;
+
+        let tag = user.tags?.find((tag) =>
+          tag.toLowerCase().includes(value.toLowerCase())
+        );
+        if (tag) return true;
```
Comment (eddiejaoude) [inline pages/search.js:54]: Please use the same coding style as the rest of the project 
```suggestion
        if (tag) {
           return true;
        }
```
```

**Comment by eddiejaoude** [comment]
Looks good 👍  Some suggestions

- use the same coding style
- write an automated test for the new feature, or write a skeleton test with `test.fixme` style
-

File: tests/profile.spec.js [hunk 2]
```diff
@@ -63,3 +63,10 @@ test.fixme("Link navigates", async ({ page }) => {
   // 3. click the link
   // 4. get the current url and should match href
 });
+
+test.fixme("redirect to search when tag clicked", async ({ page }) => {
+  // 1. redirect to profile
+  // 2. get a link and href
+  // 3. click the link
+  // 4. get the current url and should match href
+});
```
Comment (eddiejaoude) [inline tests/profile.spec.js:72]: thank you 👍

File: pages/search.js [hunk 3]
```diff
@@ -44,9 +45,16 @@ export default function Search({ users }) {
 
     if (value.length >= 3) {
       setThreeOrMore(true);
-      results = users.filter((user) =>
-        user.name.toLowerCase().includes(value.toLowerCase())
-      );
+      results = users.filter((user) => {
+        if (user.name.toLowerCase().includes(value.toLowerCase())) return true;
```
Comment (eddiejaoude) [inline pages/search.js:49]: Similar to line 54
```suggestion
        if (user.name.toLowerCase().includes(value.toLowerCase())) {
           return true;
        }
```

Comment (joeshajan) [reply pages/search.js]: Sorry @eddiejaoude this mistake will not be repeated

**Comment by joeshajan** [comment]
Can someone please explain why this test fails and how to fix this.

**Comment by joeshajan** [comment]
Hey, @eddiejaoude thank you so much for explaining where I went wrong.
I fixed it and now the tests are passing.

**Review by eddiejaoude — APPROVED** [review]
Looks great, thank you 👍

**Pull request merged**